quickwit/quickwit-proto/protos/quickwit/search.proto

```diff
@@ -111,6 +115,58 @@ message ReportSplitsRequest {
 
 message ReportSplitsResponse {}
 
+// -- ListFields -------------------
+
+message ListFieldsRequest {
+  // Optional limit query to a set of indexes.
+  repeated string index_id = 1;
+  // Optional limit query to a list of fields
+  // Wildcard expressions are supported.
+  repeated string fields = 2;
+
+  // Control if the the request will fail if split_ids contains a split that does not exist. 
+  // optional bool fail_on_missing_index = 3;
+}
+
+message LeafListFieldsRequest {
+  // The index id
+  string index_id = 1;
+  // The index uri
+  string index_uri = 2;
+  // Index split ids to apply the query on.
+  // This ids are resolved from the index_uri defined in the search_request.
+  repeated SplitIdAndFooterOffsets split_offsets = 3;
+
+  // Optional limit query to a list of fields
+  // Wildcard expressions are supported.
+  repeated string fields = 4;
+  
+  // `DocMapper` as json serialized trait.
+  string doc_mapper = 5;
+}
+
+message ListFieldsResponse {
+  repeated ListFieldsEntryResponse fields = 1;
+}
+
+message ListFieldsEntryResponse {
+  // The field name
+  string field_name = 1;
+  // The tantivy field type mapped via `to_code`.
+  uint32 field_type = 2;
```

Comment:
make it a proto enum to remove the dependency in quickwit-serve.

Reply:
I would rather reuse the existing code